musicApp/R/functions.R

```diff
@@ -167,3 +177,115 @@ getAlbumLyrics = function(albumname, artist, progressBar = FALSE)
 
   return(data)
 }
+
+
+# creating a playlist based on a user selected playlist -------------------
+
+
+# function that creates a playlist based on another playlist which is inputted by the user
+# currently the playlist is made on my own spotify profile (https://open.spotify.com/user/1117862156?si=OMhP-ANnSae9ijUFoqT_ew)
+# because i havent found out how i can let someone without a developper account log in.
+# if you want to try the function you need a playlist ID, which you can get by sharing the link to a playlist
+# (id is the letters/numbers in the url between 'playlist/' and '?si=') note that the playlist you use has to be public
+
+makePlaylist <- function(playlistID){
+
+  # create dataframe with data about the playlist and create variable with list of artists.
+  playlist = get_playlist_audio_features('Karel Veldkamp', playlistID)
+  artists = playlist %>%
+    unnest(track.artists) %>%
+    select(name, id) %>%
+    unique
+
+
+  # sample 30 artists (if n > 30)
```

Comment:
Shortly explain in a comment why exactly 30 artists? Is this just for coding in this script? As you're planning to incorporate this in a Shiny App, you could make this in a variable and let the user choose the number of artists.